sdks/go/test/regression/coders/fromyaml/fromyaml.go

```diff
@@ -0,0 +1,415 @@
+			got = elem.Elm
+		} else {
+			got = elem.Windows[0]
+		}
+
+	case "beam:coder:global_window:v1":
+		want = window.GlobalWindow{}
+		// If this is nested in an iterable, windows won't be populated.
+		if len(elem.Windows) == 0 {
+			got = window.GlobalWindow(elem.Elm.(struct{}))
+		} else {
+			got = elem.Windows[0]
+		}
+	case "beam:coder:windowed_value:v1", "beam:coder:param_windowed_value:v1":
+		// elem contains all the information, but we need to compare the element+timestamp
+		// separately from the windows, to avoid repeated expected value parsing logic.
+		pass := true
+		vs := eg.Value.(yaml.MapSlice)
+		if !diff(c.Components[0], elem, vs[0]) {
+			pass = false
+		}
+		if d := cmp.Diff(
+			mtime.FromMilliseconds(int64(vs[1].Value.(int))),
+			elem.Timestamp, cmpOpts...); d != "" {
+
+			pass = false
+		}
+		if !diff(c.Components[1], elem, vs[3]) {
+			pass = false
+		}
+		// TODO compare pane information.
+		return pass
+	case "beam:coder:row:v1":
+		fs := eg.Value.(yaml.MapSlice)
+		var rfs []reflect.StructField
+		// There are only 2 pointer examples, but they reuse field names,
+		// so we key off the proto hash to know which example we're handling.
+		ptrEg := strings.Contains(c.Payload, "51ace21c7393")
+		for _, rf := range fs {
+			name := rf.Key.(string)
+			t := nameToType[name]
+			if ptrEg {
+				t = reflect.PtrTo(t)
+			}
+			rfs = append(rfs, reflect.StructField{
+				Name: strings.ToUpper(name[:1]) + name[1:],
+				Type: t,
+				Tag:  reflect.StructTag(fmt.Sprintf("beam:\"%v\"", name)),
+			})
+		}
+		rv := reflect.New(reflect.StructOf(rfs)).Elem()
+		for i, rf := range fs {
+			setField(rv, i, rf.Value)
+		}
+
+		got, want = elem.Elm, rv.Interface()
+	default:
+		got, want = elem.Elm, eg.Value
+	}
+	if d := cmp.Diff(want, got, cmpOpts...); d != "" {
+		log.Printf("Decoding error: diff(-want,+got): %v\n", d)
+		return false
+	}
+	return true
+}
+
+// standard_coders.yaml uses the name for type indication, except for nullability.
+var nameToType = map[string]reflect.Type{
+	"str":     reflectx.String,
+	"i32":     reflectx.Int32,
+	"f64":     reflectx.Float64,
+	"arr":     reflect.SliceOf(reflectx.String),
+	"f_bool":  reflectx.Bool,
+	"f_bytes": reflect.PtrTo(reflectx.ByteSlice),
+	"f_map":   reflect.MapOf(reflectx.String, reflect.PtrTo(reflectx.Int64)),
+}
+
+func setField(rv reflect.Value, i int, v interface{}) {
+	if v == nil {
+		return
+	}
+	rf := rv.Field(i)
+	if rf.Kind() == reflect.Ptr {
+		// Ensure it's initialized.
+		rf.Set(reflect.New(rf.Type().Elem()))
+		rf = rf.Elem()
+	}
+	switch rf.Kind() {
+	case reflect.String:
+		rf.SetString(v.(string))
+	case reflect.Int32:
+		rf.SetInt(int64(v.(int)))
+	case reflect.Float64:
+		c, err := strconv.ParseFloat(v.(string), 64)
+		if err != nil {
+			panic(err)
+		}
+		rf.SetFloat(c)
+	case reflect.Slice:
+		if rf.Type() == reflectx.ByteSlice {
+			rf.Set(reflect.ValueOf([]byte(v.(string))))
+			break
+		}
+		// Value is a []interface{} with string values.
```

Comment:
Is this assumption made specifically based on the yaml file this is meant to be used with? Or is this based on something in the schema spec I'm forgetting?

Reply:
The yaml file exclusively uses lists of strings, which is why we know what type to interpret them as here.

Reply:
Gotcha. It seems a little brittle (if the yaml file ever adds an example with a list of other types then this will break), so a comment mentioning that it's due to the contents of the yaml file would be helpful if that happens.